src/components/tx/security/safenet/index.tsx

```diff
@@ -0,0 +1,148 @@
+    case 'recipient_signature':
+      return 'Recipient verification'
+    default:
+      return 'Other'
+  }
+}
+
+function _groupResultGuarantees({
+  results,
+}: Pick<SafeNetSimulationResponse, 'results'>): { display: string; success: boolean }[] {
+  const groups = results.reduce((groups, { guarantee, success }) => {
+    const display = _getGuaranteeDisplayName(guarantee)
+    return {
+      ...groups,
+      [display]: (groups[display] ?? true) && success,
+    }
+  }, {} as Record<string, boolean>)
+  return Object.entries(groups)
+    .map(([display, success]) => ({ display, success }))
+    .sort((a, b) => a.display.localeCompare(b.display))
+}
+
+function _getSafeTxHash({ safe, chainId, safeTx }: Required<SafeNetTxSimulationProps>): string {
+  return hashTypedData({
+    domain: {
+      chainId,
+      verifyingContract: safe,
+    },
+    types: {
+      SafeTx: [
+        { type: 'address', name: 'to' },
+        { type: 'uint256', name: 'value' },
+        { type: 'bytes', name: 'data' },
+        { type: 'uint8', name: 'operation' },
+        { type: 'uint256', name: 'safeTxGas' },
+        { type: 'uint256', name: 'baseGas' },
+        { type: 'uint256', name: 'gasPrice' },
+        { type: 'address', name: 'gasToken' },
+        { type: 'address', name: 'refundReceiver' },
+        { type: 'uint256', name: 'nonce' },
+      ],
+    },
+    message: { ...safeTx.data },
+  })
+}
+
+const SafeNetTxTxSimulationSummary = ({ simulation }: { simulation: SafeNetSimulationResponse }): ReactElement => {
+  if (simulation.results.length === 0) {
+    return <Typography>No SafeNet checks enabled...</Typography>
+  }
+
+  const guarantees = _groupResultGuarantees(simulation)
+
+  return (
+    <Paper variant="outlined" className={css.wrapper}>
+      {simulation.hasError && (
+        <Typography color="error" className={css.errorSummary}>
+          One or more SafeNet checks failed!
+        </Typography>
+      )}
+
+      <List>
+        {guarantees.map(({ display, success }) => (
+          <ListItem key={display}>
+            <ListItemIcon>
+              {success ? (
+                <SvgIcon component={CheckIcon} inheritViewBox fontSize="small" color="success" />
+              ) : (
+                <SvgIcon component={CloseIcon} inheritViewBox fontSize="small" color="error" />
+              )}
+            </ListItemIcon>
+            <ListItemText>{display}</ListItemText>
+          </ListItem>
+        ))}
+      </List>
+    </Paper>
+  )
+}
+
+export const SafeNetTxSimulation = ({ safe, chainId, safeTx }: SafeNetTxSimulationProps): ReactElement | null => {
+  const [dataDecoded] = useDecodeTx(safeTx)
+  const [simulate, { data: simulation, status }] = useLazySimulateSafeNetTxQuery()
+
+  useEffect(() => {
+    if (!safeTx || !dataDecoded) {
+      return
+    }
+
+    const safeTxHash = _getSafeTxHash({ safe, chainId, safeTx })
+    simulate({
+      chainId,
+      tx: {
+        safe,
+        safeTxHash,
```

Comment:
Can't the backend compute it? Feel like it's not required for simulation

Reply:
I checked and _technically_ it isn't required for the simulation (I passed in a fake hash and it worked :sweat_smile:). One reason I like it is that it gives a caching tag for the Redux toolkit `createApi`.